clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractCoordinator.java

```diff
@@ -204,23 +204,14 @@ public synchronized void ensureCoordinatorReady() {
                 time.sleep(retryBackoffMs);
             }
         }
+        // Since coordinator is set in a synchronized block in the group coordinator response callback,
+        // the future associated with the cooordinator request can be safely cleared here.
+        findCoordinatorFuture = null;
```

Comment:
Perhaps a better option for nulling `findCoordinatorFuture` is to do it in `coordinatorUnknown()` since that is called on every iteration of the poll() loop (and in the heartbeat thread).